### DISCUSSION TITLE
Disabling Image Optimization in Local and Dev Env

### DISCUSSION THREAD

**Discussion by SanjeebLama** [vercel/community#1463] [Help]
### Summary

I want to disable the image optimization in my local and dev environment in Vercel. 
I have added a line in `next.config.js` to enable next/image optimization only in the production environment,
```
  images: {
        ...(process.env.NODE_ENV !== 'production' && { unoptimized: true }), 
        minimumCacheTTL: 10000,
        domains: [
          "xyz-bucket.s3.ap-south-1.amazonaws.com",
```

 but I am encountering issues with **`View Source Images`** As it's not displaying any new source images but the source image and optimizations counts keeps increasing.

<img width="290" alt="image" src="https://user-images.githubusercontent.com/51410633/217245681-a4a692e9-37c2-482e-aaf9-1a44862c1c4b.png">
 
 **After sending Above config code**
<img width="1335" alt="image" src="https://user-images.githubusercontent.com/51410633/217246285-b348ff9a-8822-44d0-bb48-5b73736d103e.png">

**Now, I have this:**
![image](https://user-images.githubusercontent.com/51410633/217246518-c188bdec-7b08-49c7-ae80-098f5b5079e0.png)
 
 Above code has somehow disabled View Source images as it doesn't display any new urls but the count of source images and optimized images are increasing.

I have three questions that I need help with: 
1) Will the code I provided work for enabling image optimization in the **production environment only**?  If not, what steps should I take to achieve this?
2) Why am I not seeing the new source images after deploying above changes in my `next.config.js`? and 
3) Pricing documentation says that for every 1000 image it will charge $5, I don't want to go above limits. Will it automatically start charging after image optimization count crosses 5000? If yes, how to stop this?

**Comment by amyegan** [comment]
Hi Sanjeeb! You can find some strategies for [managing Image Optimization usage and costs](https://vercel.com/docs/concepts/image-optimization/managing-image-optimization-costs#) in the Vercel docs. 

Be careful about dynamic image src, height, and width values which can generate a large number of [source Images](https://vercel.com/docs/concepts/image-optimization#source-images). Image files are cached and are carried over between deployments, so a source image with the exact same URL will only be optimized one time. But, if the source image URL has **path values or query strings that change**, the image could be optimized many times.

> See the common usages of next/image listed below:
> - `import img from './img.png'; <Image src={img} />` at-most, counts as 1 source image if the image is requested during the billing period.
> - `<Image src="/logo.png" width={100} height={100} />` at-most, counts as 1 source image if the image is requested during the billing period.
> - `<Image src={'https://cdn.example.com/${path}'} width={100} height={100} />` counts as N source images - one for each uniquepath requested during the billing period.
> - `<Image src={img} loader={myCustomerFunc} />` counts as 0 source images since optimization is performed by a 3rd party image provider, not Vercel.
> - `<Image src={img} unoptimized />` counts as 0 source images since the image is never optimized.

I'm not sure about your strategy for disabling image optimization in for preview and development environments, but the [Next.js folks](https://github.com/vercel/next.js/discussions) might have some suggestions. 

Any overage will automatically be added to your bill added to your account when you exceed 5000 source images

**Reply by SanjeebLama** [reply]
Hi Amy, Thanks for the response. 

I have been monitoring my system. It looks like the code to disable image optimization in local and preview /develop environments didn't work.

I did went through the [Next.js Discussions](https://github.com/vercel/next.js/discussions) and documentation again but couldn't really find anything thats similar to the issue I am facing.

I will be disabling it for the current billing cycle because I don't want to go over my limit.

If you find anything on how to enable image optimization only in production environment, please send them towards me.
Thanks. 🙇🏻

**Reply by sra-013** [reply]
Hi Sanjeeb,

Were you able to disable it for prod? I tried but all my images stopped getting rendered. Any insights would be appreciated. Thank you!

**Reply by SanjeebLama** [reply]
Hey @sra-013,

I also faced similar problem after setting `unoptimized: true `in my next.config.js.
It's possible that your images have stopped rendering because they are still being served as "next/image". To fix this, you can add the "**unoptimized**" keyword to the next/image components.
Like:

`  <Image
                src={src}
                layout="fill"
                alt={alt}
                unoptimized
   />`
   
Also, YES after deploying my updated config file I was able to disable image optimization in production.

**Reply by sra-013** [reply]
Adding the `unoptimized` prop to the `Image` component worked! Thank you! 

> because they are still being served as "next/image"

Yes this is right, but how to bypass this? do you know? 

> was able to disable image optimization in production

Did you achieve this by using the unoptimised prop with Images or by updating your next config file?

**Reply by SanjeebLama** [reply]
From my understanding, after deploying `unoptimized: true` in next.config.js will stop optimizing your `Image` components altogether.
Then your images will stop rendering because they are served as next/image. 

There are two ways to by pass this.
1. Either replace next `Image` component with `img` tag
or
2. Add `unoptimized` prop to the next `Image` component.
